### PR TITLE
423 histogram accuracy message

### LINKED ISSUE #423: histogram: not working/better error messages
(1) UI Fix. 

- [ ] In `AddStatisticsDialog.vue`, change line 266 to:
```
 "Histogram": ["Integer", "Categorical", "Boolean"]
```

(2) Checking for errors

- [ ] Test the histogram with various data types (integer, categorical, boolean, NOT float) and check results
   - Give more context to error messages like ("NoneType is not iterable")

Use the eye fatigue test file to check across variable types (integer, categorical, boolean, NOT float). 
- for bounds/categories, actual values may be found in this file: https://github.com/opendp/dpcreator/blob/develop/server/test_data/non_dp_debug_info/info_Fatigue_data.json

---

- References: `dp_histogram_spec.py`

### PR DESCRIPTION
**Pull request by ecowan**
Closes #423

### CODE REVIEW

**Comment by ekraffmiller** [comment]
I tested  this with the EyeHeight data, using Trial variable (type integer) and Histogram. It passed validation on the Add Statistics step, but the release returned a 400 error (attached)
Screen Shot 2022-01-20 at 1 18 21 PM

File: server/opendp_apps/analysis/tools/stat_spec.py [hunk 1]
```diff
@@ -82,7 +83,11 @@ def __init__(self, props: dict):
 
         self.categories = self.variable_info.get('categories')
         self.var_type = self.variable_info.get('type')
-
+        # print("VAR TYPE", self.var_type)
+        if self.var_type == VAR_TYPE_INTEGER:
```
Comment (raprasad) [inline server/opendp_apps/analysis/tools/stat_spec.py]: This should go into the dp_histogram_spec, e.g. under `run_01_initial_handling` or another abstract method.
(fyi: I'll make this change while testing)

File: server/opendp_apps/analysis/tools/dp_histogram_spec.py [hunk 2]
```diff
@@ -100,9 +102,19 @@ def get_preprocessor(self):
...
+            toa = int
+            tia = int
+        else:
+            toa = str
+            tia = str
+
         preprocessor = (
             make_select_column(key=self.col_index, TOA=str) >>
-            make_count_by_categories(categories=self.categories, MO=L1Distance[int], TIA=str)
+            make_cast(TIA=str, TOA=toa) >>
```
Comment (raprasad) [inline server/opendp_apps/analysis/tools/dp_histogram_spec.py:116]: @ecowan error may be here? e.g. `TIA=str` instead of `make_case(TIA=tia, TOA=toa) >>`?

Comment (ecowan) [reply server/opendp_apps/analysis/tools/dp_histogram_spec.py:116]: This is here for imputing different types based on the variable info

**Review by raprasad — APPROVED** [review]
approved

**Pull request merged**